maven-compat/src/test/java/org/apache/maven/project/AbstractMavenProjectTestCase.java

```diff
@@ -140,7 +121,11 @@ protected static File getFileForClasspathResource( String resource )
     protected ArtifactRepository getLocalRepository()
         throws Exception
     {
-        return repositorySystem.createLocalRepository( getLocalRepositoryPath() );
+        ArtifactRepositoryLayout repoLayout = lookup( ArtifactRepositoryLayout.class, "legacy" );
+
+        ArtifactRepository r = repositorySystem.createArtifactRepository( "local", "file://" + getLocalRepositoryPath().getAbsolutePath(), repoLayout, null, null );
```

Comment:
This is wrong. You cannot construct a URI like that. Please do: `toPath().toUri().toASCIIString()`.

Reply:
I restored the unittest, so now it is back in its original state with the bad createArtifactRepository.
Let's fix that in a separate issue, this one is already too big.